### ISSUE TITLE
False positive: identity_conversion on macro-generated `?`

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#4081]
This is a regression in the most recent nightly because I just started seeing Travis failures related to this.

If you take the macro-generated code and paste it into main.rs, the same lint does not trigger.

```console
$ cat Cargo.toml 
[package]
name = "repro"
version = "0.0.0"
edition = "2018"

[lib]
proc-macro = true
```

```console
$ cat src/lib.rs 
extern crate proc_macro;
use proc_macro::TokenStream;

#[proc_macro]
pub fn m(_input: TokenStream) -> TokenStream {
    "fn f() -> Result<(), ()> { Ok(())?; Ok(()) }"
    .parse().unwrap()
}
```

```console
$ cat src/main.rs 
repro::m!();

fn main() {}
```

```console
$ cargo clippy
    Checking repro v0.0.0 (/git/repro)
warning: identical conversion
 --> src/main.rs:1:12
  |
1 | repro::m!();
  |            ^
  |
  = note: #[warn(clippy::identity_conversion)] on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#identity_conversion
```

```console
$ cargo clippy -V
clippy 0.0.212 (3710ec5 2019-05-11)
```

**Comment by Manishearth** [comment]
Yeah, we had to change how the macro checks work because of some changes wrt if desugaring. I'll insert a stronger check for that lint.
